docs/cookbook/payments/custom-field-on-refund-payment.rst

```diff
@@ -0,0 +1,607 @@
+        }
+    }
+
+And handler:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\ProcessManager;
+
+    use Doctrine\ORM\EntityManagerInterface;
+    use Sylius\Component\Core\Model\OrderInterface;
+    use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+    use Sylius\RefundPlugin\Entity\RefundPaymentInterface;
+    use Sylius\RefundPlugin\Event\RefundPaymentGenerated;
+    use Sylius\RefundPlugin\Event\UnitsRefunded;
+    use Sylius\RefundPlugin\Factory\RefundPaymentFactoryInterface;
+    use Sylius\RefundPlugin\ProcessManager\UnitsRefundedProcessStepInterface;
+    use Sylius\RefundPlugin\Provider\RelatedPaymentIdProviderInterface;
+    use Sylius\RefundPlugin\StateResolver\OrderFullyRefundedStateResolverInterface;
+    use Symfony\Component\Messenger\MessageBusInterface;
+    use Webmozart\Assert\Assert;
+
+    final class RefundPaymentProcessManager implements UnitsRefundedProcessStepInterface
+    {
+        /** @var OrderFullyRefundedStateResolverInterface */
+        private $orderFullyRefundedStateResolver;
+
+        /** @var RelatedPaymentIdProviderInterface */
+        private $relatedPaymentIdProvider;
+
+        /** @var RefundPaymentFactoryInterface */
+        private $refundPaymentFactory;
+
+        /** @var OrderRepositoryInterface */
+        private $orderRepository;
+
+        /** @var EntityManagerInterface */
+        private $entityManager;
+
+        /** @var MessageBusInterface */
+        private $eventBus;
+
+        public function __construct(
+            OrderFullyRefundedStateResolverInterface $orderFullyRefundedStateResolver,
+            RelatedPaymentIdProviderInterface $relatedPaymentIdProvider,
+            RefundPaymentFactoryInterface $refundPaymentFactory,
+            OrderRepositoryInterface $orderRepository,
+            EntityManagerInterface $entityManager,
+            MessageBusInterface $eventBus
+        ) {
+            $this->orderFullyRefundedStateResolver = $orderFullyRefundedStateResolver;
+            $this->relatedPaymentIdProvider = $relatedPaymentIdProvider;
+            $this->refundPaymentFactory = $refundPaymentFactory;
+            $this->orderRepository = $orderRepository;
+            $this->entityManager = $entityManager;
+            $this->eventBus = $eventBus;
+        }
+
+        public function next(UnitsRefunded $unitsRefunded): void
+        {
+            /** @var OrderInterface|null $order */
+            $order = $this->orderRepository->findOneByNumber($unitsRefunded->orderNumber());
+            Assert::notNull($order);
+
+            $refundPayment = $this->refundPaymentFactory->createWithDataAndDate(
+                $order,
+                $unitsRefunded->amount(),
+                $unitsRefunded->currencyCode(),
+                RefundPaymentInterface::STATE_NEW,
+                $unitsRefunded->paymentMethodId(),
+                $unitsRefunded->getScheduledAt()
+            );
```

Comment:
Perhaps this could be part of `RefundUnitsHandler` and we could simplify events in general

Reply:
It will be more complicated than I initially thought, as creating RefundPayments and CreditMemos is based on these events. A change like this would force us to change the logic behind creating RefundPayments and CreditMemos in proper order. It would also change the other [cookbook](https://docs.sylius.com/en/latest/cookbook/payments/refund-process.html) so I would like to stay with events in RefundPlugin 1.0.